fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/StatisticsCalculator.java

```diff
@@ -1350,17 +1350,23 @@ private Void computeCTEConsume(Operator node, ExpressionContext context, int cte
 
         // None children, may force CTE, use the statistics of producer
         Optional<Statistics> produceStatisticsOp = optimizerContext.getCteContext().getCTEStatistics(cteId);
-        Preconditions.checkState(produceStatisticsOp.isPresent(),
+        // Outside of memo, produceStatisticsOp may be empty, because rbo-phase join reordering required by
+        // TablePruneRboRule only handle multi-join sub-queries, if the sub-query contains CTEConsumeOperator
+        // without children and the CTEProduceOperator is outside of this sub-query, the statistics of
+        // CTEProduceOperator is not estimated yet.
+        Preconditions.checkState(optimizerContext.getMemo().getGroups().isEmpty() ||
+                        produceStatisticsOp.isPresent(),
                 "cannot obtain cte statistics for %s", node);
-        Statistics produceStatistics = produceStatisticsOp.get();
+
         Statistics.Builder builder = Statistics.builder();
         for (ColumnRefOperator ref : columnRefMap.keySet()) {
             ColumnRefOperator produceRef = columnRefMap.get(ref);
-            ColumnStatistic statistic = produceStatistics.getColumnStatistic(produceRef);
+            ColumnStatistic statistic = produceStatisticsOp.map(stat -> stat.getColumnStatistic(produceRef))
+                    .orElse(ColumnStatistic.unknown());
```

Comment:
I think column statistic shouldn't be unknown.

Reply:
It is only used by RboTablePruneRule outside of memo,  no needs to do statistics inferring.